packages/dashboard-core-plugins/src/PandasPlugin.tsx

```diff
@@ -1,28 +1,22 @@
-import {
-  assertIsDashboardPluginProps,
-  DashboardPluginComponentProps,
-  useDashboardPanel,
-} from '@deephaven/dashboard';
-import { useApi } from '@deephaven/jsapi-bootstrap';
+import { DashboardPanelProps } from '@deephaven/dashboard';
+import { forwardRef, useMemo } from 'react';
 import { PandasPanel } from './panels';
 import useHydrateGrid from './useHydrateGrid';
 
-export function PandasPlugin(
-  props: DashboardPluginComponentProps
-): JSX.Element | null {
-  assertIsDashboardPluginProps(props);
-  const dh = useApi();
-  const hydrate = useHydrateGrid();
+export const PandasPlugin = forwardRef(
+  (props: DashboardPanelProps, ref: React.Ref<PandasPanel>) => {
+    const hydrate = useHydrateGrid<DashboardPanelProps>();
+    const { localDashboardId } = props;
+    const hydratedProps = useMemo(
+      () => hydrate(props, localDashboardId),
+      [hydrate, props, localDashboardId]
+    );
 
-  useDashboardPanel({
-    dashboardProps: props,
-    componentName: PandasPanel.COMPONENT,
-    component: PandasPanel,
-    supportedTypes: dh.VariableType.PANDAS,
-    hydrate,
-  });
+    // eslint-disable-next-line react/jsx-props-no-spreading
```

Comment:
We should probably just disable this rule globally. We spread props all the time.

Reply:
Idk I think I agree with the rule of being explicit usually. We mostly use it when we're just blindly passing through props and there's no alternative.

I think TS checks spread props in JSX now, but not sure if it checks for excess props.

Here's the rule. Can also open a ticket to disable this if you think we should.

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md

Reply:
Eh let's not worry about it for now.